tests/unit/test_middleware/test_middleware_handling.py

```diff
@@ -102,13 +103,22 @@ class JSONRequest:
     def post_handler(data: JSONRequest) -> JSONRequest:
         return data
 
-    with caplog.at_level(logging.INFO):
+    if sys.version_info < (3, 13):
+        with caplog.at_level(logging.INFO):
+            client = create_test_client(
+                route_handlers=[post_handler], middleware=[MiddlewareProtocolRequestLoggingMiddleware]
+            )
+            response = client.post("/", json={"name": "moishe zuchmir", "age": 40, "programmer": True})
+            assert response.status_code == 201
+            assert "test logging" in caplog.text
+    else:
         client = create_test_client(
             route_handlers=[post_handler], middleware=[MiddlewareProtocolRequestLoggingMiddleware]
         )
         response = client.post("/", json={"name": "moishe zuchmir", "age": 40, "programmer": True})
         assert response.status_code == 201
-        assert "test logging" in caplog.text
+        log = capsys.readouterr()
+        assert "test logging" in log.err
```

Comment:
I've tried everything to get these to behave more consistently.  For now, this is working, but I think it should be revisited.  However, let's not hold up because of this.